spec/api_spec.rb

```diff
@@ -137,6 +140,96 @@
     end
   end
 
+  context 'messages' do
+    let(:device) {
+      VCR.use_cassette('set_device_attributes') do
+        api.set_device_attributes(devices)
+      end
+    }
+    let(:user) {
+      VCR.use_cassette('set_user_attributes') do
+        api.set_user_attributes(users)
+      end
+    }
+    let(:spock) {
+      VCR.use_cassette('spock') do
+        api.set_user_attributes(
+          [{
+             user_id: "spock",
+             first_name: "Spock",
+             gender: "m",
+             email: "spock@ufop.org",
+             create_date: "2018-01-01".to_date
+          }]
+        )
+      end
+    }
+
+    describe '#send_message' do
+      it "requires message_id" do
+        expect { api.send_message(user_id: user_id) }.to raise_error(ArgumentError, "missing keyword: message_id")
+      end
+
+      it "requires user_id or user_ids" do
+        expect { api.send_message(message_id: message_id) }.to raise_error(ArgumentError, "missing keyword: user_id or user_ids")
+      end
+
+      it "succesfully sends a message to Leanplum's API" do
+        user
+        response = VCR.use_cassette('send_message') do
+          api.send_message(message_id: message_id, user_id: user_id)
+        end
+        expect(response.size).to eq 1
+        response = response.first
+        expect(response['success']).to eq true
+        expect(response['result']).to eq "Message sent"
+        expect(response['messagesSent']).to eq 1
+        expect(response['error']).to be_blank
+      end
+
+      it "succesfully sends a message to Leanplum's API using device_id" do
```

Comment:
what about an unsuccessful case? like a bad user id or a message id that does not exist?

Reply:
@beingmattlevy added!